### ISSUE TITLE
Build Issue with Master branch and Visual Studio 2017

### ISSUE THREAD

**Issue by gra-moore** [facebook/rocksdb#2262]
When building using cmake and visual studio 2017. The following compile error is raised:
```
Error	C2678	binary '-': no operator found which takes a left-hand operand of type 'const rocksdb::autovector<rocksdb::TransactionID,8>::iterator_impl<const rocksdb::autovector<rocksdb::TransactionID,8>,const unsigned __int64>' (or there is no acceptable conversion)	rocksdb	d:\programs\vs2017\vc\tools\msvc\14.10.25017\include\xutility	695	
```
The same error is generated 6 times, 3 from rocksdb and 3 from rocksdb-shared

To reproduce:
```
mkdir build
cd build
cmake -G "Visual Studio 15 Win64" ..
```
Then:
```
msbuild rocksdb.sln
```
Visual Studio Version is: 15.1 26403.7

xutility is located: `d:\Programs\VS2017\VC\Tools\MSVC\14.10.25017\include\xutility`

**Comment by adamretter** [comment]
Any ideas @yuslepukhin?

**Comment by yuslepukhin** [comment]
We have not switched the compiler internally but I will take a look when I get a chance.

**Comment by dibyendumajumdar** [comment]
Hi, I am trying to build the latest 5.4.6 release and facing the same issue.

**Comment by sagar0** [comment]
@dibyendumajumdar this fix is not in 5.4.6. If you plan to build with Visual Studio 2017, please try building master, for now. I can confirm that master's building fine with VS 2017.
